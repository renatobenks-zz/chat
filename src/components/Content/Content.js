// @flow

import * as React from 'react';
import { Helmet } from 'react-helmet';
import styled from 'styled-components';

const Wrapper = styled.div`
  flex: 1;
  padding: ${props => props.theme.padding};
`;

type Props = {
  title: ?string,
  children: React.Element<{}>,
};

const Content = ({ children, title }: Props) => {
  return (
    <Wrapper>
      <Helmet>{title && <title>{title}</title>}</Helmet>
      {children}
    </Wrapper>
  );
};

export default Content;
