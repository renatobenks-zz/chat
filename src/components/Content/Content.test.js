import React from 'react';
import renderer from 'react-test-renderer';

import Content from './Content';

const title = 'Mocked title';

it('renders content children with title', () => {
  expect(
    renderer.create(
      <Content title={title}>
        <span>Awesome title with: {title}</span>
      </Content>,
    ),
  ).toMatchSnapshot();
});

it('renders content children with no title', () => {
  expect(
    renderer.create(
      <Content>
        <span>No title</span>
      </Content>,
    ),
  ).toMatchSnapshot();
});
