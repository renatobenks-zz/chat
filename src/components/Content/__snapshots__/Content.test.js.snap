// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders content children with no title 1`] = `
<div
  className="sc-bdVaJa cWsHuj"
>
  <span>
    No title
  </span>
</div>
`;

exports[`renders content children with title 1`] = `
<div
  className="sc-bdVaJa cWsHuj"
>
  <span>
    Awesome title with: 
    Mocked title
  </span>
</div>
`;
